File: list_models.py
import os
import requests
import json

def main():
    # Read the .env file
    env_vars = {}
    try:
        with open('c:\\Users\\mmdig_q4f4o3g\\Documents\\Workspace\\task-flutter\\.env', 'r') as f:
            for line in f:
                if '=' in line:
                    k, v = line.strip().split('=', 1)
                    env_vars[k.strip()] = v.strip()
    except Exception as e:
        print("Error reading .env:", e)
        return

    api_key = env_vars.get('GEMINI_API_KEY')
    if not api_key:
        print("No GEMINI_API_KEY found.")
        return

    url = f"https://generativelanguage.googleapis.com/v1beta/models?key={api_key}"
    response = requests.get(url)
    
    if response.status_code == 200:
        models = response.json().get('models', [])
        print("Available models:")
        for m in models:
            name = m.get('name')
            if 'gemini' in name.lower():
                print(name)
    else:
        print(f"Error {response.status_code}: {response.text}")

if __name__ == '__main__':
    main()

File: list_models.dart
import 'dart:convert';
import 'dart:io';

void main() async {
  final envFile = File('.env');
  if (!await envFile.exists()) {
    print('No .env file found');
    return;
  }
  
  final lines = await envFile.readAsLines();
  String apiKey = '';
  for (final line in lines) {
    if (line.startsWith('GEMINI_API_KEY=')) {
      apiKey = line.split('=')[1].trim();
      break;
    }
  }

  if (apiKey.isEmpty) {
    print('No API key found in .env');
    return;
  }

  final client = HttpClient();
  final request = await client.getUrl(Uri.parse('https://generativelanguage.googleapis.com/v1beta/models?key=$apiKey'));
  final response = await request.close();
  
  final responseBody = await response.transform(utf8.decoder).join();
  if (response.statusCode == 200) {
    final Map<String, dynamic> data = jsonDecode(responseBody);
    final models = data['models'] as List<dynamic>;
    for (final model in models) {
      if (model['name'].toString().contains('gemini')) {
        print(model['name']);
      }
    }
  } else {
    print('Error ${response.statusCode}: $responseBody');
  }
  client.close();
}
